src/scenes/Partners/Edit/EditPartner.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { ComponentType, useMemo } from 'react';
+import { Except, Merge } from 'type-fest';
+import {
+  displayFinancialReportingType,
+  FinancialReportingTypeList,
+  GQLOperations,
+  PartnerTypeList,
+  UpdatePartner,
+} from '../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../components/Dialog/DialogForm';
+import {
+  CheckboxField,
+  EnumField,
+  SubmitError,
+  TextField,
+} from '../../../components/form';
+import { UserField, UserLookupItem } from '../../../components/form/Lookup';
+import { ExtractStrict, many, Many } from '../../../util';
+import { PartnerDetailsFragment } from '../Detail/PartnerDetail.generated';
+import { useUpdatePartnerMutation } from './EditPartner.generated';
+
+interface PartnerFormValues {
+  partner: Merge<
+    UpdatePartner,
+    {
+      pointOfContactId?: UserLookupItem;
+    }
+  >;
+}
+
+export type EditablePartnerField = ExtractStrict<
+  keyof UpdatePartner,
+  // Add more fields here as needed
+  | 'pointOfContactId'
+  | 'globalInnovationsClient'
+  | 'pmcEntityCode'
+  | 'active'
+  | 'types'
+  | 'financialReportingTypes'
+  | 'address'
+>;
+
+type EditPartnerProps = Except<
+  DialogFormProps<PartnerFormValues>,
+  'onSubmit' | 'initialValues'
+> & {
+  partner: PartnerDetailsFragment;
+  editFields?: Many<EditablePartnerField>;
+};
+interface EngagementFieldProps {
+  props: {
+    name: string;
+  };
+  hide?: boolean;
+}
+
+const fieldMapping: Record<
+  EditablePartnerField,
+  ComponentType<EngagementFieldProps>
+> = {
+  pointOfContactId: ({ props }) => (
+    <UserField {...props} label="Point of Contact" />
+  ),
+  globalInnovationsClient: ({ props }) => (
+    <CheckboxField {...props} label="Global Innovations Client" />
+  ),
+  active: ({ props }) => <CheckboxField {...props} label="Active" />,
+  pmcEntityCode: ({ props }) => (
+    //TODO: add in validation 3 uppercase chars
+    <TextField {...props} label="PMC Entity Code" />
+  ),
+  types: ({ props }) => (
+    <EnumField
+      multiple
+      label="Types"
+      options={PartnerTypeList}
+      layout="two-column"
+      {...props}
+    />
+  ),
+  //TODO: fix hard crash and missing initialValues
```

Comment:
Max rerender crashing here, and also for some reason the initialValues aren't populating.

Reply:
Fixed by Carson's fix!

Reply:
This isn't populating because we're giving a radio input, which takes a single string value, an array, which is what our `UpdatePartner` input wants us to use. This seems like a mismatch between the UI and the schema to me. Are users supposed to be able to submit multiple `financialReportingTypes` for a partner? Then this should be a checkbox field. Otherwise, `UpdatePartner` should take an Enum, not an array of Enums

Reply:
Yes multiple here. This only defines what's allowed to be selected in the partnership object

Reply:
Got it, changed to multiple and fixed.  I noticed that in UpdatePartnership the `financialReportingType` is singular, is that intended?

Reply:
No it should be multiple. I had to point that out on the API PR too, so it must've been missed in the update.